cudamapper/src/index_cache.cuh

```diff
@@ -67,10 +68,26 @@ public:
     ~IndexCacheHost()                           = default;
 
     /// \brief Discards previously cached query Indices, creates new Indices and copies them to host memory
-    void update_query_cache(const std::vector<IndexDescriptor>& descriptors_of_indices_to_cache);
+    ///
+    /// Expected usage pattern is to call update_query_cache() with certain index descriptors and then immediately retrieve some of them.
+    /// To avoid immediately copying back those indices from host to device it is possible to specify descriptors_of_indices_to_keep_on_device
+    //  which will be copied to host, but also kept on device until retrieved for the first time using get_index_from_query_cache
+    ///
+    /// \param descriptors_of_indices_to_cache descriptors on indices to keep in host memory
+    /// \param descriptors_of_indices_to_keep_on_device descriptors of indices to keep in device memory in addition to host memory until retrieved for the first time
+    void update_query_cache(const std::vector<IndexDescriptor>& descriptors_of_indices_to_cache,
```

Comment:
I got a bit confused by the name `update_*_cache`. It took me a while to realize, that this does not just add elements to the cache, but actually generates the indices to be cached.
Would `generate_*_cache_content()` be a better name?

Reply:
reads better now. thanks!